src/BuildScriptGenerator/DotNetCore/ProjectFileProviders/ProbeAndFindProjectFileProvider.cs

```diff
@@ -63,26 +69,77 @@ public string GetRelativePathToProjectFile(RepositoryContext context)
 
             var webAppProjects = new List<string>();
             var azureFunctionsProjects = new List<string>();
+            var azureBlazorWasmProjects = new List<string>();
             var allProjects = new List<string>();
+            bool functionProjectExists = false;
+            bool webAppProjectExists = false;
+            bool blazorWasmProjectExists = false;
+
             foreach (var file in projectFiles)
             {
                 allProjects.Add(file);
-                if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                if (ProjectFileHelpers.IsAzureBlazorWebAssemblyProject(sourceRepo, file))
                 {
-                    webAppProjects.Add(file);
+                    azureBlazorWasmProjects.Add(file);
+                    blazorWasmProjectExists = true;
                 }
                 else if (ProjectFileHelpers.IsAzureFunctionsProject(sourceRepo, file))
                 {
                     azureFunctionsProjects.Add(file);
+                    functionProjectExists = true;
+                }
+                else if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                {
+                    webAppProjects.Add(file);
+                    webAppProjectExists = true;
                 }
             }
 
-            projectFile = GetProject(webAppProjects);
-            if (projectFile == null)
+            // Assumption: some build option "--apptype" will be passed to oryx
+            // which will indicate if the app is an azure function type or static type app
+
+            // If there are multiple projects, we will look for --appty to detect corresponding project.
+            // for example azurefunction and blazor both projects can reside
+            // at the same repo, so more than 2 csprojs will be found. Now we will
+            // look for --apptype value to determine which project needs to be built
+            if (_options != null
+                && _options.AppType != null)
+            {
+                _logger.LogInformation($"ProbeAndProjectFileProvider: {Constants.AppType} is set to {_options.AppType}");
+
+                if (functionProjectExists && _options.AppType.ToLower().Contains("functions"))
+                {
+                    projectFile = GetProject(azureFunctionsProjects);
+                }
+                else if (blazorWasmProjectExists
+                    && _options.AppType.ToLower().Contains("static-sites"))
+                {
+                    projectFile = GetProject(azureBlazorWasmProjects);
+                }
+                else
+                {
+                    _logger.LogDebug($"Invalid value {_options.AppType} for env:{Constants.AppType}. Currently, supported values are 'functions', 'blazor-wasm', 'static-sites'");
```

Comment:
`env:{Constants.AppType}` => `{SettingsKeys.AppType}`...also the supported values can be put in a constants file which can referenced here and as well as in BuildCommand.IsValidInput